src/main/java/com/minecolonies/coremod/inventory/InventoryCitizen.java

```diff
@@ -837,7 +838,7 @@ else if (this.canMergeStacks(this.getStackInSlot(40), itemStackIn))
     private boolean canMergeStacks(final ItemStack stack1, final ItemStack stack2)
     {
         return !stack1.isEmpty() && InventoryCitizen.stackEqualExact(stack1, stack2) && stack1.isStackable()
```

Comment:
Here is still an isEmpty() check.
Please replace that with the ItemStackUtil one.